koku/masu/external/downloader/azure/azure_report_downloader.py

```diff
@@ -57,18 +57,12 @@ def get_processing_date(
         tracing_id (str): The tracing id
     """
     dh = DateHelper()
-    date_format = "%Y-%m-%d %H:%M:%S"
-    time_interval = "UsageDateTime"
-    try:
-        pd.read_csv(local_file, usecols=[time_interval], nrows=1)
-    except ValueError:
-        time_interval = "Date"
-        date_format = "%Y-%m-%d"
-        try:
-            pd.read_csv(local_file, usecols=[time_interval], nrows=1)
-        except ValueError:
-            time_interval = "date"
-            date_format = "%m/%d/%Y"
+    time_interval = None
+    columns = pd.read_csv(local_file, nrows=0)
+    for interval in ["UsageDateTime", "Date", "date"]:
+        if interval in columns:
+            time_interval = interval
+            break
```

Comment:
We should probably look at the columns, specifically:
```suggestion
    columns = pd.read_csv(local_file, nrows=0).columns
    for interval in ["UsageDateTime", "Date", "date"]:
        if interval in columns:
            time_interval = interval
            break
```
Alternatively, you could do some a set operation (intersection) to get the column you're looking for:
```suggestion
time_interval = pd.read_csv(local_file, nrows=0).columns.intersection({"UsageDateTime", "Date", "date"})[0]
```
though this method is likely more difficult to comprehend at first glance.

Reply:
Plenty of options here. :smile: So originally I was avoiding loading all the columns but since we have not implemented usecols _yet_ that might not actually matter here for now. That being said regarding casings, I have seen another report with lower `usagedatestart`. With that in mind we _could_ do something like this:

`cols = ["usagedatetime", "date"]`
`df = pandas.read_csv(local_file, usecols=lambda x: x.lower() in cols, nrows=0)`
 `df.columns[0]`

That way we could handle all the column header case types. :+1: